src/data/reports/glass-admin/GLASSDataMaintenanceDefaultRepository.ts

```diff
@@ -0,0 +1,492 @@
+                                id: true,
+                                level: true,
+                            },
+                        },
+                    },
+                    filter: {
+                        level: {
+                            eq: "1",
+                        },
+                    },
+                },
+            })
+            .getData();
+
+        const orgUnitsWithChildren = _(userOrgUnits)
+            .flatMap(ou => {
+                const res = flattenNodes(flattenNodes(organisationUnits).filter(res => res.id === ou)).map(
+                    node => node.id
+                );
+                return res;
+            })
+            .uniq()
+            .value();
+
+        return orgUnitsWithChildren;
+    }
+
+    private async getUserOrgUnits(): Promise<string[]> {
+        return (await this.api.get<{ organisationUnits: Ref[] }>("/me").getData()).organisationUnits.map(ou => ou.id);
+    }
+
+    private async getCountriesOutsideNARegion(): Promise<string[]> {
+        const { organisationUnits: kosovoOu } = await this.api.metadata
+            .get({
+                organisationUnits: {
+                    fields: {
+                        id: true,
+                    },
+                    filter: {
+                        name: {
+                            eq: "Kosovo",
+                        },
+                        level: {
+                            eq: "3",
+                        },
+                        "parent.code": {
+                            eq: "NA",
+                        },
+                    },
+                },
+            })
+            .getData();
+
+        const { organisationUnits } = await this.api.metadata
+            .get({
+                organisationUnits: {
+                    fields: {
+                        id: true,
+                    },
+                    filter: {
+                        level: {
+                            eq: "3",
+                        },
+                        "parent.code": {
+                            ne: "NA",
+                        },
+                    },
+                    paging: false,
+                },
+            })
+            .getData();
+
+        return _.concat(organisationUnits, kosovoOu).map(orgUnit => orgUnit.id);
+    }
+
+    private uploadNewATC(year: string, atcItems: ATCItem[]): ATCItem[] {
+        const atcYears = atcItems.map(atcItem => _.parseInt(atcItem.year));
+        const currentVersionYear = Math.max(...atcYears);
+
+        const newItem: ATCItem = {
+            year: year,
+            version: "1",
+            uploadedDate: new Date().toISOString(),
+            currentVersion: _.parseInt(year) > currentVersionYear,
+            previousVersion: false,
+        };
+
+        return [...atcItems, newItem];
+    }
+
+    private patchATCVersion(atcItems: ATCItem[], selectedItems: ATCItemIdentifier[]) {
```

Comment:
Also here take into account what I have said about `currentVersion` and `previousVersion` above, because `currentVersion` may be changed to true here